Content/Grid/grid.gd

```diff
@@ -99,16 +121,32 @@ func update_grid():
 		add_score(matches.size())
 	
 	if matches.size() == 0:
-		if settled:
+		if is_board_settled():
 			done_updating = true
+			
+		# Add tiles to the game
 		for x in grid_width:
-			if (add_tile(x, 0)):
-				done_updating = false
-	else:
-		settled = false
+			# Get the amount of tiles to be added
+			var new_tile_count: int = 0
+			for y in grid_height:
+				if not is_instance_valid(grid[y][x]):
+					new_tile_count += 1
+					done_updating = false
+				else:
+					break
```

Comment:
not important but we can do
```
if is_instance_valid(grid[y][x]): break
...
```
to reduce indentation

Reply:
absolutely !

Reply:
fixed in this commit: 78b371fb66d6f3e2c6e35e7b368ea2ec5851c97a